src/App.js

```diff
@@ -35,21 +37,76 @@ const styles = theme => ({
 });
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      authContext: {
```

Comment:
If we pull this out into a separate `AuthProvider` component we can flatten this state and remove the nested authContext object, which makes updates simpler